### DISCUSSION TITLE
How to add custom file system？

### DISCUSSION THREAD

**Discussion by JySongWithZhangCe** [duckdb/duckdb#8602] [Q&A]
I would like to add some extension of the distributed file system for duckdb, any suggestions? Thanks♪(･ω･)ﾉ

**Comment by Mause** [comment]
We have an example of this in the httpfs extension (https://github.com/duckdb/duckdb/tree/master/extension/httpfs) and https://github.com/duckdb/duckdb/blob/master/tools/pythonpkg/src/pyfilesystem.cpp

**Reply by JySongWithZhangCe** [reply]
Thank you for the advice! I will try it!

**Comment by Tishj** [comment]
Do you mean you want to add a custom FileSystem so duckdb can use it when a filepath is issued that is recognized by the filesystem?
i.e `read_csv_auto('ftp://file_path')`

If so, then you might want to have a look at the HTTPFS extensions code, as it does just that, for `http`

**Reply by JySongWithZhangCe** [reply]
Thanks♪(･ω･)ﾉ !!!
